Add support for pluggable Custom Presto Authenticators (#24111)
prestodb/presto: presto-docs/src/main/sphinx/develop/presto-authenticator.rst

===========================
Custom Presto Authenticator
===========================

> Presto supports authentication via a custom presto authenticator

Reviewer: ```suggestion
Presto supports authentication through a custom Presto authenticator
```
Don't use `via`. See the GitLab documentation style guide word list entry for via for discussion and alternate working suggestions, if the suggested `through` isn't a good fit for you.

that validates the request and creates a principal.

Implementation
--------------

> ``PrestoAuthenticatorFactory`` is responsible for creating a

Reviewer: ```suggestion
``PrestoAuthenticatorFactory`` creates a
```

that validates the request and returns a ``Principal``, which is then
authorized by the :doc:`system-access-control`.

The implementation of ``PrestoAuthenticatorFactory`` must be wrapped
as a plugin and installed on the Presto cluster.

Configuration
-------------

After a plugin that implements ``PrestoAuthenticatorFactory`` has been
installed on the coordinator, it is configured using an
``etc/presto-authenticator.properties`` file. All of the
properties other than ``presto-authenticator.name`` are specific to the
``PrestoAuthenticatorFactory`` implementation.

The ``presto-authenticator.name`` property is used by Presto to find a
registered ``PrestoAuthenticatorFactory`` based on the name returned by
``PrestoAuthenticatorFactory.getName()``. The remaining properties are
passed as a map to ``PrestoAuthenticatorFactory.create()``.

Example configuration file:

.. code-block:: none

    presto-authenticator.name=custom-authenticator
    custom-property1=custom-value1
    custom-property2=custom-value2

Additionally, the coordinator must be configured to use custom authentication
and have HTTPS enabled.

> Below property needs to be added to the coordinator's ``config.properties`` file:

Reviewer: ```suggestion
Add the property shown below to the coordinator's ``config.properties`` file:
```
